cluster-autoscaler/cloudprovider/gce/mig_info_provider.go

```diff
@@ -444,3 +446,25 @@ func (c *cachingMigInfoProvider) GetListManagedInstancesResults(migRef GceRef) (
 	c.cache.SetListManagedInstancesResults(migRef, listManagedInstancesResults)
 	return listManagedInstancesResults, nil
 }
+
+func createInstancesState(actionsSummary *gce.InstanceGroupManagerActionsSummary) map[cloudprovider.InstanceState]int64 {
+	state := map[cloudprovider.InstanceState]int64{
+		cloudprovider.InstanceCreating: 0,
+		cloudprovider.InstanceDeleting: 0,
+		cloudprovider.InstanceRunning:  0,
+	}
+	state[getInstanceState("ABANDONING")] += actionsSummary.Abandoning
```

Comment:
The thing that worries me about this approach is that if a new action is added, we will not count those instances at all and have no idea.

So we need to either
a) make sure that no actions will ever be added to the list of possible vm actions
b) just get the counts for "interesting actions" ("CREATING", "RECREATING", "CREATING_WITHOUT_RETRIES", "ABANDONING", "DELETING") from actionsSummary and calculate the running instances as targetSize - creating. This requires confirmation that targetSize = #instances_in_all_states - #instances_deleting - #instances_abandoning

OTOH, do we need the exact number of running instances? Maybe we just want to keep the total instances count? That number will AFAICT be equal to targetSize + #instances_deleting + #instances_abandoning, although this needs to be confirmed as well

Reply:
Implemented option b)